modules/ggl-yaml/src/yaml_decode.c

```diff
@@ -203,12 +203,33 @@ GglError ggl_yaml_decode_destructive(
     static pthread_mutex_t mtx = PTHREAD_MUTEX_INITIALIZER;
     GGL_MTX_SCOPE_GUARD(&mtx);
 
+    GGL_LOGT(
+        "%s received yaml content: %.*s", __func__, (int) buf.len, buf.data
+    );
+
     static yaml_parser_t parser;
-    yaml_parser_initialize(&parser);
+    if (!yaml_parser_initialize(&parser)) {
```

Comment:
As a todo, we should figure out a way to not have a parser which calls malloc.

Reply:
I'm not crazy enough to write our own YAML parser

Though could be an option if we restrict to a subset of YAML.